pytensor/tensor/subtensor.py

```diff
@@ -2486,15 +2532,104 @@ def gen_num(typen):
         return code
 
     def c_code(self, node, name, input_names, output_names, sub):
-        if numpy_version < "1.8.0" or using_numpy_2:
-            raise NotImplementedError
-
         x, y, idx = input_names
-        out = output_names[0]
+        [out] = output_names
         copy_of_x = self.copy_of_x(x)
         params = sub["params"]
         fail = sub["fail"]
 
+        x_, y_, idx_ = node.inputs
+        y_dtype = y_.type.dtype_specs()[1]
```

Comment:
```suggestion
        y_ctype = y_.type.dtype_specs()[1]
```

Makes it a bit more clear what's going on here (I had to look up what dtype_specs does)

Reply:
done